src/Datadog.Trace/AsyncLocalScopeManager.cs

```diff
@@ -22,18 +22,16 @@ public void Close(Scope scope)
         {
             var current = _activeScope.Get();
 
-            if (current != scope)
+            if (current != null && current == scope)
             {
-                Log.Warn("Specified scope is not the active scope.");
+                // replace active scope with its parent,
+                // but only if it's the scope we were expecting
+                _activeScope.Set(current.Parent);
```

Comment:
Under what circumstances would this not be the scope we expected?

Reply:
> Under what circumstances would this not be the scope we expected?

If a scope that was just closed is _not_ the active scope.

The driver for this change is Web API. Web API can raise the `EndRequest` event on a different thread, so we don't have access to the active scope at all. We end up writing both log entries for every Web API request, which is unnecessary.

Your question made rething this. Since this is a normal occurrence, I'm removing the log message entirely.